src/lib/executors/Node.ts

```diff
@@ -875,13 +876,46 @@ export default class Node extends Executor<NodeEvents, Config, NodePlugins> {
         }
 
         testTask
-          .then(() => {
+          .then(async () => {
             if (!this._sessionSuites) {
               return;
             }
-            return this._loadFunctionalSuites().then(
-              () => (testTask = this._runRemoteTests())
-            );
+
+            const allSessions = this._sessionSuites;
+            let suites = allSessions;
+            let remainingAttempts = 1 + (this.config.functionalRetries || 0);
+
+            await this._loadFunctionalSuites();
+            while (remainingAttempts && suites.length) {
+              remainingAttempts--;
+              if (suites.length !== allSessions.length) {
+                this.log(
+                  'reattempting',
```

Comment:
We ideally shouldn't say 'reattempting' on the first run.

Reply:
suites.length and allSessions.length will be equal on the first run. Suites are filtered later as successful ones are removed.

Reply:
Ah, good point